### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by moray95**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        Some(config) => tonic_builder.compile_with_config(config, &protos, &includes),
+    }
+    .map_err(|e| TonicBufBuildError::new("error running tonic build", e.into()))
+}
+
+pub fn compile_from_buf(
+    tonic_builder: tonic_build::Builder, config: Option<prost_build::Config>,
+) -> Result<(), TonicBufBuildError> {
+    let export_dir = tempdir();
+    defer! {
```
Comment (can-arikan) [inline src/lib.rs:73]: does this work as in golang ?

File: src/buf.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
+use std::path::Path;
+
+use serde::{Deserialize, Serialize};
+
+use crate::error::TonicBufBuildError;
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
```
Comment (can-arikan) [inline src/buf.rs:7]: as far as I see, deserialization is not used in this project

File: src/buf.rs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        let f = std::fs::File::open(file)
+            .map_err(|e| TonicBufBuildError::new(&format!("failed to read {}", file), e.into()))?;
+
+        let buf: BufYaml = serde_yaml::from_reader(&f)
+            .map_err(|e| TonicBufBuildError::new(&format!("failed to deserialize {}", file), e.into()))?;
+        Ok(buf)
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
```
Comment (can-arikan) [inline src/buf.rs:23]: as far as I see, deserialization is not used in this project

File: src/lib.rs (hunk 1 above)
Comment (can-arikan) [reply src/lib.rs:73]: if so, the comment on this block could be more clear. Hence as far as I understand, this code block tries to clean the temp directory when the execution of the main ends. I could not understand the import relation which is mentioned in the comment.

File: src/buf.rs (hunk 3 above)
Comment (moray95) [reply src/buf.rs]: True, fixed

**Pull request merged**